sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/CompactBlockProcessor.kt

```diff
@@ -193,25 +201,33 @@ class CompactBlockProcessor(
-                    val summary = if (noWorkDone) "Nothing to process: no new blocks to download or scan" else "Done processing blocks"
-                    consecutiveChainErrors.set(0)
-                    val napTime = calculatePollInterval()
-                    twig("$summary${if (result == ERROR_CODE_FAILED_ENHANCE) " (but there were enhancement errors! We ignore those, for now. Memos in this block range are probably missing! This will be improved in a future release.)" else ""}! Sleeping for ${napTime}ms (latest height: ${currentInfo.networkBlockHeight}).")
-                    delay(napTime)
-                } else {
-                    if (consecutiveChainErrors.get() >= RETRIES) {
-                        val errorMessage = "ERROR: unable to resolve reorg at height $result after ${consecutiveChainErrors.get()} correction attempts!"
-                        fail(CompactBlockProcessorException.FailedReorgRepair(errorMessage))
-                    } else {
-                        handleChainError(result)
+                when (result) {
+                    BlockProcessingResult.Reconnecting -> {
+                        val napTime = calculatePollInterval(true)
+                        twig("Unable to process new blocks because we are disconnected! Attempting to reconnect in ${napTime}ms")
+                        delay(napTime)
+                    }
+                    BlockProcessingResult.NoBlocksToProcess, BlockProcessingResult.FailedEnhance -> {
+                        val noWorkDone =
+                            currentInfo.lastDownloadRange?.isEmpty() ?: true && currentInfo.lastScanRange?.isEmpty() ?: true
+                        val summary = if (noWorkDone) {
+                            "Nothing to process: no new blocks to download or scan"
+                        } else {
+                            "Done processing blocks"
+                        }
+                        consecutiveChainErrors.set(0)
+                        val napTime = calculatePollInterval()
+                        twig("$summary${if (result == BlockProcessingResult.FailedEnhance) " (but there were enhancement errors! We ignore those, for now. Memos in this block range are probably missing! This will be improved in a future release.)" else ""}! Sleeping for ${napTime}ms (latest height: ${currentInfo.networkBlockHeight}).")
+                        delay(napTime)
+                    }
+                    is BlockProcessingResult.Error -> {
+                        val napTime = calculatePollInterval(true)
+                        twig("Unable to process new blocks because they failed validation! Attempting to reconnect in ${napTime}ms")
+                        delay(napTime)
+                    }
+                    is BlockProcessingResult.Success -> {
+                        // Do nothing. We are done.
                     }
-                    consecutiveChainErrors.getAndIncrement()
```

Comment:
I think that the consecutive errors count is not being updated as it was. Was this moved somewhere else?

Reply:
This should be resolved now.